src/zenml/orchestrators/dag_runner.py

```diff
@@ -168,6 +174,14 @@ def _run_node(self, node: str) -> None:
         """
         self._prepare_node_run(node)
 
+        # Check if execution should continue (e.g., check for cancellation)
+        if self.check_fn and not self.check_fn():
```

Comment:
If this function is something that cancels the entire dag runner, wouldn't it make sense to simply run it once here, if it returns that it should be canceled, set a boolean and not rerun every time?